File: services/intelligence-api/app/schemas/kpi.py
"""
KPI schemas — executive, country-level, and sector-level KPI models.

Every KPI: named, valued, trended, categorized, explained.
"""
from pydantic import BaseModel, Field
from .scenario import LanguageVariant, GCCCountryCode, SectorCode


class KPITrend(BaseModel):
    """Trend metadata for a KPI."""
    direction: str = Field(pattern="^(up|down|stable)$")
    change_pct: float
    previous_value: float = 0.0


class ExecutiveKPI(BaseModel):
    """Top-level executive KPI."""
    id: str
    name: LanguageVariant
    value: float
    unit: str
    trend: KPITrend
    category: str = Field(pattern="^(economic|risk|opportunity|governance|infrastructure|financial)$")
    explanation: LanguageVariant


class CountryKPI(BaseModel):
    """Per-country KPI."""
    country_code: GCCCountryCode
    kpi_id: str
    name: LanguageVariant
    value: float
    unit: str
    trend: KPITrend
    explanation: LanguageVariant


class SectorKPI(BaseModel):
    """Per-sector KPI."""
    sector_code: SectorCode
    kpi_id: str
    name: LanguageVariant
    value: float
    unit: str
    trend: KPITrend
    explanation: LanguageVariant


class KPIDashboard(BaseModel):
    """Full KPI output for a scenario."""
    scenario_id: str
    executive_kpis: list[ExecutiveKPI]
    country_kpis: list[CountryKPI]
    sector_kpis: list[SectorKPI]
    headline: LanguageVariant

File: apps/executive-dashboard/src/types/kpi.ts
/**
 * KPI Types — exact mirror of schemas/kpi.py
 *
 * Source: services/intelligence-api/app/schemas/kpi.py
 */

import type { LanguageVariant, GCCCountryCode, SectorCode } from "./scenario";

export interface KPITrend {
  direction: "up" | "down" | "stable";
  change_pct: number;
  previous_value: number;
}

export interface ExecutiveKPI {
  id: string;
  name: LanguageVariant;
  value: number;
  unit: string;
  trend: KPITrend;
  category: "economic" | "risk" | "opportunity" | "governance" | "infrastructure" | "financial";
  explanation: LanguageVariant;
}

export interface CountryKPI {
  country_code: GCCCountryCode;
  kpi_id: string;
  name: LanguageVariant;
  value: number;
  unit: string;
  trend: KPITrend;
  explanation: LanguageVariant;
}

export interface SectorKPI {
  sector_code: SectorCode;
  kpi_id: string;
  name: LanguageVariant;
  value: number;
  unit: string;
  trend: KPITrend;
  explanation: LanguageVariant;
}

export interface KPIDashboard {
  scenario_id: string;
  executive_kpis: ExecutiveKPI[];
  country_kpis: CountryKPI[];
  sector_kpis: SectorKPI[];
  headline: LanguageVariant;
}
